oracle/block.go

```diff
@@ -295,9 +306,24 @@ func (b *FullBlock) MevRewardInWei() (*big.Int, bool, string) {
 		log.Fatal("could not get tx sender: ", err)
 	}
 
+	whitelistedBuilders, found := WhitelistedBuilders[b.ChainId]
+	if !found {
+		log.Fatal("Chain not found in whitelisted builders: ", b.ChainId)
+	}
+
 	// Mev rewards are sent in the last tx. This tx sender
-	// matches the fee recipient of the protocol
-	if utils.Equals(b.GetFeeRecipient(), sender.String()) {
+	// matches the fee recipient of the protocol.
+	// We also consider a MEV reward if the tx comes from a whitelisted builder. This
+	// is rare, but has happened: https://beaconcha.in/slot/9444748
+	if utils.Equals(b.GetFeeRecipient(), sender.String()) ||
+		utils.IsIn(sender.String(), whitelistedBuilders) {
```

Comment:
Just to be sure, the method where this `if` is, `MevRewardInWei()`, is only accessed when the pool has been rewarded in the block right? Here: https://github.com/dappnode/mev-sp-oracle/blob/main/oracle/onchain.go#L920

In other worlds, will we be printing these logs everytime this builder builds a block?

Reply:
Indeed! Fixed: 8affe8edc00dbe5cc9aef8c01a4b2edf9d96e406
great catch.